src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -649,6 +649,23 @@ public function dumpFile($filename, $content)
     }
 
     /**
+     * Atomically appends content to an existing file.
+     *
+     * @param string $filename The file for which to append content
+     * @param string $content  The content to append
+     *
+     * @throws IOException If the file is not writable
+     */
+    public function appendToFile($filename, $content)
+    {
+        if (false !== $originalContent = @file_get_contents($filename)) {
```

Comment:
Im not an filesystem expert..   but the tmp file in the middle feels like a poor man solution to me..  so perhaps `LOCK_EX` is the way to go indeed for `dumpeFile` as well.